core/src/test/java/com/novoda/noplayer/exoplayer/ExoPlayerFacadeTest.java

```diff
@@ -0,0 +1,319 @@
+
+@RunWith(Enclosed.class)
+public class ExoPlayerFacadeTest {
+
+    private static final long TWO_MINUTES_IN_MILLIS = 120000;
+    private static final long TEN_MINUTES_IN_MILLIS = 600000;
+
+    private static final int TEN_PERCENT = 10;
+
+    private static final boolean IS_PLAYING = true;
+    private static final boolean PLAY_WHEN_READY = true;
+    private static final boolean DO_NOT_PLAY_WHEN_READY = false;
+    private static final boolean RESET_POSITION = true;
+    private static final boolean DO_NOT_RESET_STATE = false;
+
+    private static final ContentType ANY_CONTENT_TYPE = ContentType.DASH;
+
+    public static class GivenVideoNotLoaded extends Base {
+
+        private static final VideoPosition ANY_POSITION = VideoPosition.fromMillis(1000);
+
+        @Rule
+        public ExpectedException thrown = ExpectedException.none();
+
+        @Test
+        public void whenResetting_thenReleasesUnderlyingPlayer() {
+
+            facade.release();
+
+            verify(exoPlayer, never()).release();
+        }
+
+        @Test
+        public void whenLoadingVideo_thenAddsPlayerEventListener() {
+
+            facade.loadVideo(uri, ANY_CONTENT_TYPE, exoPlayerForwarder);
+
+            verify(exoPlayer).addListener(exoPlayerForwarder.exoPlayerEventListener());
+        }
+
+        @Test
+        public void whenLoadingVideo_thenSetsVideoDebugListener() {
+
+            facade.loadVideo(uri, ANY_CONTENT_TYPE, exoPlayerForwarder);
+
+            verify(exoPlayer).setVideoDebugListener(exoPlayerForwarder.videoRendererEventListener());
+        }
+
+        @Test
+        public void givenMediaSource_whenLoadingVideo_thenPreparesInternalExoPlayer() {
+            MediaSource mediaSource = givenMediaSource();
+
+            facade.loadVideo(uri, ANY_CONTENT_TYPE, exoPlayerForwarder);
+
+            verify(exoPlayer).prepare(mediaSource, RESET_POSITION, DO_NOT_RESET_STATE);
+        }
+
+        @Test
+        public void whenSelectingAudioTrack_thenDelegatesToTrackSelector() {
+            PlayerAudioTrack audioTrack = mock(PlayerAudioTrack.class);
+
+            facade.selectAudioTrack(audioTrack);
+
+            verify(trackSelector).selectAudioTrack(audioTrack);
+        }
+
+        @Test
+        public void whenGettingAudioTracks_thenDelegatesToTrackSelector() {
+
+            facade.getAudioTracks();
+
+            verify(trackSelector).getAudioTracks();
+        }
+
+        @Test
+        public void whenQueryingIsPlaying_thenReturnsFalse() {
+
+            boolean isPlaying = facade.isPlaying();
+
+            assertThat(isPlaying).isFalse();
+        }
+
+        @Test
+        public void whenQueryingPlayheadPosition_thenThrowsIllegalStateException() {
+            thrown.expect(IllegalStateException.class);
+
+            facade.getPlayheadPosition();
+        }
+
+        @Test
+        public void whenQueryingMediaDuration_thenThrowsIllegalStateException() {
+            thrown.expect(IllegalStateException.class);
```

Comment:
On the `AndroidMediaPlayerFacade` we are asserting the expected thrown with greater granularity. Should we do the same here?